js/src/main/scala/scorex/crypto/hash/Platform.scala

```diff
@@ -1,40 +1,84 @@
 package scorex.crypto.hash
 
-import scorex.crypto.BouncycastleJs
+import typings.nobleHashes.blake2Mod.BlakeOpts
+import typings.nobleHashes.mod.blake2b
+import typings.nobleHashes.sha256Mod.sha256
+import typings.nobleHashes.utilsMod
 
-import scala.scalajs.js
+import scala.scalajs.js.typedarray.Uint8Array
 
+/** JS platform specific implementation of methods.
+  * When shared code is compiled to JS, this implementation is used.
+  *
+  * The JS implementation is based on type wrappers generated by ScalablyTyped for the
+  * @noble/hashes library. (See configuration in build.sbt.)
+  *
+  * @see jvm/src/main/scala/scorex/crypto/hash/Platform.scala for JVM implementation
+  */
 object Platform {
 
-  type Digest = BouncycastleJs.Digest
-  
+  /** Represents abstract digest from @noble.
+    * See createBlake2bDigest, createSha256Digest methods.
+    */
+  type Digest = utilsMod.Hash[_]
+
+  private def bytesToShorts(xs: Array[Byte]): Array[Short] =
+    xs.map(x => (x & 0xFF).toShort)
+
+  private def uint8ArrayToBytes(jsShorts: Uint8Array): Array[Byte] = {
+    jsShorts.toArray[Short].map(x => x.toByte)
+  }
+
+  /** Creates an implementation of the cryptographic hash function Blakbe2b.
+    *
+    * @param bitSize the bit size of the digest
+    * @return the digest implementation
+    */
   def createBlake2bDigest(bitSize: Int): Digest = {
-    val bc = BouncycastleJs.bouncyCastle
-    val digest = bc.createBlake2bDigest(bitSize)
-    digest
+    val opts = BlakeOpts().setDkLen(bitSize / 8)
+    blake2b.create(opts)
   }
 
+  /** Creates an implementation of the cryptographic hash function SHA-256.
+    *
+    * @return the digest implementation
+    */
   def createSha256Digest(): Digest = {
-    val bc = BouncycastleJs.bouncyCastle
-    val digest = bc.createSha256Digest()
-    digest
+    sha256.create()
   }
 
+  /** Update the message digest with a single byte.
+    *
+    * @param digest the digest to be updated
+    * @param b      the input byte to be entered.
+    */
   def updateDigest(digest: Digest, b: Byte): Unit = {
-    digest.updateByte(b)
+    digest.update(Uint8Array.of((b & 0xFF).toShort))
```

Comment:
Why (b & 0xFF).toShort ? Seems like you maybe updating digest with 2 bytes input instead of 1 before. Do you have equivalence test for that?

Reply:
Uint8Array has elements of type Short in Scala to represent UInt8 (0-255).
And .update expects Input
`type Input = js.typedarray.Uint8Array | String`, so this is correct.
There as test vectors in Blake2b256Specification and ShaSpecification which both belong to shared code and executed for both platforms.